storage/lrucache/capacity/capacityLRUCache.go

```diff
@@ -0,0 +1,252 @@
+package capacity
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+var log = logger.GetOrCreate("storage/lrucache/capacity")
+
+// CapacityLRU implements a non thread safe LRU Cache with a max capacity size
+type CapacityLRU struct {
+	lock                   sync.Mutex
+	size                   int
+	maxCapacityInBytes     int64
+	currentCapacityInBytes int64
+	evictList              *list.List
+	items                  map[interface{}]*list.Element
+}
+
+// entry is used to hold a value in the evictList
+type entry struct {
+	key   interface{}
+	value interface{}
+	size  int64
+}
+
+// NewCapacityLRU constructs an CapacityLRU of the given size with a byte size capacity
+func NewCapacityLRU(size int, byteCapacity int64) (*CapacityLRU, error) {
+	if size < 1 {
+		return nil, storage.ErrCacheSizeInvalid
+	}
+	if byteCapacity < 1 {
+		return nil, storage.ErrCacheCapacityInvalid
+	}
+	c := &CapacityLRU{
+		size:               size,
+		maxCapacityInBytes: byteCapacity,
+		evictList:          list.New(),
+		items:              make(map[interface{}]*list.Element),
```

Comment:
this can be really slow in a full environment. Might change *list.Element to a binary trie at least :)

Reply:
Will postpone this change. The capacityLRU implementation is just a roll-off of the simple lruCache that was proven to be working with what we have. Added TODO

Reply:
agreed. Can come as improvement - it is fast enough right now.